integration/spark/src/main/scala/org/apache/spark/sql/CarbonCountStar.scala

```diff
@@ -55,34 +67,39 @@ case class CarbonCountStar(
     val (job, tableInputFormat) = createCarbonInputFormat(absoluteTableIdentifier)
     CarbonInputFormat.setQuerySegment(job.getConfiguration, carbonTable)
 
-    // get row count
-    var rowCount = CarbonUpdateUtil.getRowCount(
-      tableInputFormat.getBlockRowCount(
-        job,
-        carbonTable,
-        CarbonFilters.getPartitions(
-          Seq.empty,
-          sparkSession,
-          TableIdentifier(
-            carbonTable.getTableName,
-            Some(carbonTable.getDatabaseName))).map(_.asJava).orNull, false),
-      carbonTable)
+    var totalRowCount: Long = 0
+    if (predicates.nonEmpty) {
```

Comment:
what if it is non partition table ?
a. we need to check is it partition table also ?
b. what if predicate is not partitioned column ? we need to have that check also ?

Reply:
@ajantha-bhat pls check the mothd  isPurePartitionPrune in this pr...

Reply:
ok